daml-assistant/daml-helper/src/DamlHelper.hs

```diff
@@ -126,18 +130,227 @@ withJar jarPath args a = do
+                [ ""
+                , "WARNING: da.yaml SDK version " <> versionToString projSdkVersion <> " is too old for the new"
+                , "assistant, so daml.yaml will use SDK version " <> versionToString newProjSdkVersion <> " instead."
+                , ""
+                ]
+
+        projSource :: T.Text <- getField "source"
+        projParties :: [T.Text] <- getField "parties"
+        projName :: T.Text <- getField "name"
+        projScenario :: T.Text <- getField "scenario"
+
+        BS.writeFile (projectRoot </> projectConfigName) . Y.encodePretty yamlConfig $ Y.object
+            [ ("sdk-version", Y.String (versionToText newProjSdkVersion))
+            , ("name", Y.String projName)
+            , ("source", Y.String projSource)
+            , ("scenario", Y.String projScenario)
+            , ("parties", Y.array (map Y.String projParties))
+            , ("version", Y.String "1.0.0")
+            , ("exposed-modules", Y.array [Y.String "Main"])
+            , ("dependencies", Y.array [Y.String "daml-prim", Y.String "daml-stdlib"])
+            ]
+
+
+        putStrLn ("Done! Please verify " <> projectConfigRel)
+        exitSuccess
+
+
+    -- case 7
+    putStrLn ("Generating " <> projectConfigRel)
+
+    currentSdkVersion <- getSdkVersion
+
+    projectFiles <- listFilesRecursive targetFolder
+    let isMainDotDaml = (== "Main.daml") . takeFileName
+        sourceM = listToMaybe (filter isMainDotDaml projectFiles)
+        source = fromMaybe "daml/Main.daml" sourceM
+        name = takeFileName (dropTrailingPathSeparator targetFolderAbs)
+
+    BS.writeFile (targetFolder </> projectConfigName) . Y.encodePretty yamlConfig $ Y.object
+        [ ("sdk-version", Y.String (T.pack currentSdkVersion))
+        , ("name", Y.String (T.pack name))
+        , ("source", Y.String (T.pack source))
+        , ("scenario", Y.String "Main:mainScenario")
+        , ("parties", Y.array [Y.String "Alice", Y.String "Bob"])
+        , ("version", Y.String "1.0.0")
+        , ("exposed-modules", Y.array [Y.String "Main"])
+        , ("dependencies", Y.array [Y.String "daml-prim", Y.String "daml-stdlib"])
+        ]
+
+    putStr $ unlines
+        [ "Initialized project " <> name
+        , "Done! Please verify " <> projectConfigRel
+        ]
+
+
+    where
+        legacyConfigName = "da.yaml"
+
+        findDamlProjectRoot :: FilePath -> IO (Maybe FilePath)
+        findDamlProjectRoot = findAscendantWithFile projectConfigName
+
+        findDaProjectRoot :: FilePath -> IO (Maybe FilePath)
+        findDaProjectRoot = findAscendantWithFile legacyConfigName
+
+        findAscendantWithFile :: FilePath -> FilePath -> IO (Maybe FilePath)
+        findAscendantWithFile filename path =
+            findM (\p -> doesFileExist (p </> filename)) (ascendants path)
+
+        -- why don't any good filepath libraries have something like this?
+        escapePath :: FilePath -> FilePath
+        escapePath = concatMap $ \c ->
```

Comment:
I tend to hack it up with `show` and then remove the first/last `"` characters...